cardano-testnet/src/Testnet/Filepath.hs

```diff
@@ -0,0 +1,51 @@
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Testnet.Filepath
+  ( TmpAbsolutePath(..)
+  , makeDbDir
+  , makeLogDir
+  , makeSocketDir
+  , makeSprocket
+  , makeTmpBaseAbsPath
+  ) where
+
+import           Prelude
+
+import           System.FilePath
+
+import           Hedgehog.Extras.Stock.IO.Network.Sprocket (Sprocket (..))
+
+
+
+
+makeSprocket
+  :: TmpAbsolutePath
+  -> String -- ^ node name
+  -> Sprocket
+makeSprocket tmpAbsPath node
+  = Sprocket (makeTmpBaseAbsPath tmpAbsPath) (makeSocketDir tmpAbsPath </> node)
+
+-- Temporary path used at runtime
+newtype TmpAbsolutePath = TmpAbsolutePath
+  { unTmpAbsPath :: FilePath
+  } deriving (Eq, Show)
```

Comment:
I think having `Monoid` and `IsString` would be useful for this.

Reply:
I think `Semigroup` should be enough. For easier concatenation, for example:
```haskell
let prefix = "/tmp/tmpfile" :: TmpAbsolutePath
in prefix <> ".1234.tmp"
```

Reply:
I'll add `IsString` but I'd rather be explicit and use `</>` when concatenating `FilePath`s